src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,111 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { getPostDetails } from '../../api/posts';
+import { getPostComments, removePostComment } from '../../api/comments';
+
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
+export const PostDetails = ({ selectedPostId }) => {
+  const [post, setPost] = useState(null);
+  const [comments, setComments] = useState(null);
+  const [commentsVisibility, setCommentsVisibility] = useState(false);
+
+  useEffect(() => {
+    loadData();
+  }, []);
+
+  useEffect(() => {
+    loadData();
+  }, [selectedPostId]);
+
+  const loadData = () => {
+    setPost(null);
+
+    getPostDetails(selectedPostId)
+      .then((result) => {
+        setComments(null);
+        setPost(result.data);
+
+        return result.data;
+      })
+      .then((postFromServer) => {
+        getPostComments()
+          .then((commentsFromServer) => {
+            setComments(commentsFromServer.data
+              .filter(comment => postFromServer.id === comment.postId));
+          });
+      });
```

Comment:
that's the place where `async await` looks a lot better
also, why exactly you need to load posts every time when you load comments?
try to avoid unnecessary requests